Library/ViewModels/SelectCurrencyViewModelTests.swift

```diff
@@ -124,7 +124,14 @@ internal final class SelectCurrencyViewModelTests: TestCase {
     self.updateCurrencyDidSucceed.assertValueCount(0)
     self.selectCellAtIndex.assertValues([0])
 
-    withEnvironment(apiService: MockService(changeCurrencyError: .invalidInput)) {
+    let errorEnvelope = ErrorEnvelope(
+      errorMessages: ["Something went wrong."],
+      ksrCode: nil,
+      httpCode: 1,
+      exception: nil
+    )
+    
+    withEnvironment(apiService: MockService(changeCurrencyResult: .failure(errorEnvelope))) {
```

Comment:
Not sure if you created the errorEnvelope like that on purpose but you can always do `.invalidJSON` or a few other static templates we have in place.

Reply:
Ah ok, no that's good to know for next time. I'll replace it here as well.

Reply:
Ah wait, this was created intentionally because the custom error message is check below in one of the test cases "something went wrong".

Reply:
Cool, I don't think the string being returned is too important because ultimately we just want to get an ErrorEnvelope back.

Reply:
In this case I think you are satisfying the test but the behavior of the application doesn't match it. For instance, if I abort the graph request an error will be thrown but rather than it saying "Something went wrong", it will say JSONDecoder...

This isn't a problem with your PR specifically, we need to update line 59 of `ApolloClient+RAC.swift` to throw a more readable error rather than `JSONDecoderError: foo bar`

Reply:
^^ I updated this line in my PR to send "Something went wrong".

Reply:
ok, just not sure why we need to update line 59 of `ApolloClient` to return "Something went wrong", it just takes the error and wraps it in an enum `.couldNotDecodeJSON` and passes back an error envelope that contains the message `error.localizedDescription`. Don't we want to get back the real raw error message instead of a generic string?

Reply:
Let's sync on a call and avoid further back and forth here.